internal/safefile/safe_object_directory.go

```diff
@@ -0,0 +1,108 @@
+package safefile
+
+import (
+	"path/filepath"
+	"syscall"
+	"unicode/utf16"
+	"unsafe"
+)
+
+const globalNTPath = "\\Global??"
+const bufferSize = 1024
+
+// ntWidePath converts a golang string to a wide string
+func ntWidePath(path string) ([]uint16, error) {
+	path = filepath.Clean(path)
+	fspath := filepath.FromSlash(path)
+	path16 := utf16.Encode(([]rune)(fspath))
+	if len(path16) > 32767 {
+		return nil, syscall.ENAMETOOLONG
+	}
+	return path16, nil
+}
+
+// EnumerateNTGlobalObjectDirectory queries `globalNTPath` for all entires and
+// returns the resulting entry names as a string slice.
+func EnumerateNTGlobalObjectDirectory() ([]string, error) {
```

Comment:
`safefile` seems like a strange place for this, as this package is mostly about safe filesystem operations, and this code is for enumerating NT objects. Is this code only used in one place? If so, should we just make it private in that package?

Reply:
Yeah this is really only called in the device-util utility. I wanted to avoid duplicating constants though. I'm working on refactoring this.

Reply:
I pushed my refactoring change as a separate commit in case I need to revert it. Thoughts?